### PR TITLE
persistent data and mixed version fixes

### PR DESCRIPTION
**Pull request by nutbunnies**
- fixes issue with leaving data for subsequent runs
- changes ensure_running() to use nodetool status
- added nodetool drain to destroy() if leave_data=true
- changes wait_for_compaction to run nodetool commands local to each node
- removed superfluous whitespace (let me know if this should be moved to another PR)

### CODE REVIEW

File: tool/cstar_perf/tool/benchmark.py [hunk 1]
```diff
@@ -45,7 +45,7 @@
 
 def bootstrap(cfg=None, destroy=False, leave_data=False, git_fetch=True):
     """Deploy and start cassandra on the cluster
-
```
Comment (aboudreault) [inline tool/cstar_perf/tool/benchmark.py]: Yep, I usually prefer these changes to go in a dedicated PR (or committed directly in master..), since it might bloat the current PR when reviewing.

File: tool/cstar_perf/tool/benchmark.py [hunk 2]
```diff
@@ -234,65 +234,49 @@ def wait_for_compaction(nodes=None, check_interval=30, idle_confirmations=3,
...
-                        "Failed to connect via nodetool {consecutive_connection_errors} times in a row.".format(
-                        consecutive_connection_errors=consecutive_connection_errors))
-            for node in list(nodes):
-                if pattern.search(output[node]):
-                    nodes.remove(node)
+            results = execute(cstar.multi_nodetool, cmd="compactionstats")
+            for node, output in results.iteritems():
+                if pattern.search(output.strip()):
+                    nodes.discard(node)
+
```
Comment (aboudreault) [inline tool/cstar_perf/tool/benchmark.py]: What happens now if nodetool got a ConnectionRefused?

Comment (aboudreault) [inline tool/cstar_perf/tool/benchmark.py]: kk, it is the inverse, nodetool  is now ran on each nodes. should be good..

**Comment by aboudreault** [comment]
LGTM! I would prefer this to be merged after my DSE support branch since this is fresh in your mind so you can rebase easily. @EnigmaCurry Do you think it will be possible or we need this now? In any case, I think you should remove the *whitespaces cleanup* since it will make the rebase job harder for either of us who has to do it.

**Comment by EnigmaCurry** [comment]
Yes, let's do the dse branch first, I've got a copy of prod up as a staging
server so I can easily test both today.

**Comment by nutbunnies** [comment]
I've reverted the whitespace changes and will rebase/squish into master when dse is done

**Comment by aboudreault** [comment]
Ready to rebase. Note that the old fab_cassandra.py file has been splitted (fab_common.py and fab_cassandra.py)

**Comment by EnigmaCurry** [comment]
#48 has been merged

**Comment by nutbunnies** [comment]
This should be ready to go now, but it can't run upgrading to a 3.0 cluster due to CASSANDRA-10360.  Example json test config:
```
{
  "revisions": [
    {
      "revision": "apache/cassandra-2.1",
      "cluster_name": "updating_cluster",
      "label": "base cassandra-2.1",
      "java_home": "~/fab/jvms/jdk1.7.0_72"
    },
    {
      "revision": "apache/cassandra-2.1",
      "revision_override": {"apache/cassandra-2.2": ["test_cluster_01"] },
      "cluster_name": "updating_cluster",
      "label": "cassandra-2.1 with single cassandra-2.2 node",
      "java_home": "~/fab/jvms/jdk1.7.0_72"
    },
    {
      "revision": "apache/cassandra-2.2",
      "cluster_name": "updating_cluster",
      "label": "upgraded cassandra-2.2",
      "java_home": "~/fab/jvms/jdk1.8.0_45"
    },
    {
      "revision": "apache/cassandra-2.2",
      "revision_override": {"apache/cassandra-3.0": ["test_cluster_01"] },
      "cluster_name": "updating_cluster",
      "label": "upgraded cassandra-2.2 with single cassandra-3.0 node",
      "java_home": "~/fab/jvms/jdk1.8.0_45"
    },
    {
      "revision": "apache/cassandra-3.0",
      "cluster_name": "updating_cluster",
      "label": "upgraded cassandra-3.0",
      "java_home": "~/fab/jvms/jdk1.8.0_45"
    }
  ],
  "title": "Rolling Upgrade Cluster Performance",
  "log": "stats.cassandra_21_22_30_upgrades.json",
  "operations": [
    {"type": "nodetool", "command": "version"},
    {"type": "nodetool", "command": "upgradesstables"},
    {"type": "nodetool", "command": "status"},
    {
      "type": "stress",
      "command": "user ops\\(insert=100,latest=100,range=1\\) n=1M profile=https://gist.githubusercontent.com/belliottsmith/5fd2cca8b9aaaf2d83aa/raw/3ac7f962735cf0d4a0d32c9474870cd9f6e38bb8/trades.yaml -rate threads=50 -pop seq=1..1M contents=SORTED read-lookback=uniform\\(1..1M\\) -insert visits=fixed\\(100K\\) revisit=uniform\\(1..1M\\)",
      "wait_for_compaction": true
    }
  ],
  "leave_data": true
}
```

**Comment by aboudreault** [comment]
ah, forgot about that one. will check asap

**Comment by nutbunnies** [comment]
@aboudreault I have another merge issue with node sha check that I'm hacking on -- after I get it in give another look

**Comment by nutbunnies** [comment]
@aboudreault / @EnigmaCurry : last of the merge issues are fixed - this should be good to review (again) now

**Comment by EnigmaCurry** [comment]
LGTM
